src/Norway.js

```diff
@@ -0,0 +1,292 @@
+    var scrollLeft = document.documentElement.scrollLeft || document.body.scrollLeft;
+
+    return {
+      x: parentRect.x + point[0] + scrollLeft,
+      y: parentRect.y + point[1] + scrollTop
+    };
+  }
+
+  getCountyCenter(countyIndex) {
+    const selection = this.getCounty(countyIndex);
+    return this.getCenter(selection);
+  }
+
+  fadeInTooltip() {
+    d3.select(".tooltip")
+      .transition()
+      .duration(200)
+      .style("opacity", 0.9);
+  }
+
+  positionTooltip(text, x, y) {
+    d3.select(".tooltip")
+      .html(text)
+      .style("left", x + "px")
+      .style("top", y + "px");
+  }
+
+  showCountyTooltip(countyIndex, event) {
+    this.fadeInTooltip();
+
+    const { x, y } = this.getCountyCenter(countyIndex);
+    const text = this.getCountyName(countyIndex);
+
+    this.positionTooltip(text, x, y);
+
+    // Propagate event
+    const { onMouseOver } = this.props;
+    onMouseOver(event, this.getCountyName(countyIndex));
+  }
+
+  hideTooltip(event) {
+    d3.select(".tooltip")
+      .transition()
+      .duration(500)
+      .style("opacity", 0);
+    // Propagate event
+    const { onMouseOut } = this.props;
+    onMouseOut(event);
+  }
+
+  hideCountyTooltip(countyIndex, event) {
+    this.hideTooltip(event);
+    console.log("Hide Tooltip : " + this.getCountyName(countyIndex));
+  }
+
+  /**
+   *  Municipality
+   */
+  getMunicipalityCenter(municipalityIndex) {
+    const selection = this.getMunicipality(municipalityIndex);
+    return this.getCenter(selection);
+  }
+
+  getMunicipality(municipalityIndex) {
+    const key = "#" + this.makeId("municipality", municipalityIndex);
+    return d3.select(key);
+  }
+
+  getMunicipalityNumber(municipalityIndex) {
+    return this.state.municipalities[municipalityIndex].properties.ID_1;
+  }
+
+  getMunicipalityName(municipalityIndex) {
+    return this.state.municipalities[municipalityIndex].properties.NAME_2;
+  }
+
+  handleMunicipalityClick(municipalityIndex, event) {
+    console.log("Clicked : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  showMunicipalityTooltip(municipalityIndex, event) {
+    this.fadeInTooltip();
+
+    const { x, y } = this.getMunicipalityCenter(municipalityIndex);
+    const text = this.getMunicipalityName(municipalityIndex);
+
+    this.positionTooltip(text, x, y);
+
+    // Propagate event
+    const { onMouseOver } = this.props;
+    onMouseOver(event, this.getMunicipalityName(municipalityIndex));
+
+    console.log("Show Tooltip : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  hideMunicipalityTooltip(municipalityIndex, event) {
+    this.hideTooltip(event);
+    console.log("Hide Tooltip : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  render() {
+    const { id, width, height } = this.props;
+    return (
+      <div>
+        <svg id={id} width={width} height={height} viewBox={"0 0 " + width + " " + height} onClick={e => this.handleClick(e)}>
+          <g className="counties">
```

Comment:
I find it interesting here that you're using React mostly to render the SVG instead of letting d3 do it.